ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/forms/CollectionSelection.tsx

```diff
@@ -0,0 +1,207 @@
+            availableScopes.push(initialReportScope);
+        }
+
+        // This is inefficient due to the multiple loops and the fact that we are already tracking
+        // uniqueness for the _server side_ values, but need to do it twice to handle possible client
+        // side values. However, 'N' should be small here and we are memoizing the result.
+        const sorted = sortBy(availableScopes, ({ name }) => name.toLowerCase());
+        return uniqBy(sorted, 'id');
+    }, [data, createdCollections, initialReportScope]);
+
+    // This makes sure that if a collection was deleted then we clear the scopeId
+    useEffect(() => {
+        const selectedCollection = sortedCollections.find(
+            (collection) => collection.id === selectedScopeId
+        );
+        if (!selectedCollection) {
+            onChange('');
+        }
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [data]);
+
+    function onToggleCollectionModal() {
+        setIsCollectionModalOpen((current) => !current);
+    }
+
+    function onScopeChange(_id, selection) {
+        onToggle(false);
+        onChange(selection);
+    }
+
+    let selectLoadingVariant: SelectProps['loadingVariant'];
+
+    if (isFetchingNextPage) {
+        selectLoadingVariant = 'spinner';
+    } else if (!isEndOfResults) {
+        selectLoadingVariant = {
+            text: 'View more',
+            onClick: () => fetchNextPage(),
+        };
+    }
+
+    return (
+        <>
+            <Flex alignItems={{ default: 'alignItemsFlexEnd' }}>
+                <FlexItem>
+                    <FormLabelGroup
+                        className="pf-u-mb-md"
+                        isRequired
+                        label="Configure report scope"
+                        fieldId="scopeId"
+                        touched={isLegacyReportScopeSelected ? { scopeId: true } : {}}
+                        errors={
+                            isLegacyReportScopeSelected
+                                ? { scopeId: 'Choose a new collection to use as the report scope' }
+                                : {}
+                        }
+                    >
+                        <Select
+                            id="scopeId"
+                            onSelect={onScopeChange}
+                            selections={isLegacyReportScopeSelected ? '' : selectedScopeId}
+                            placeholderText="Select a collection"
+                            variant={SelectVariant.typeahead}
+                            isOpen={isOpen}
+                            onToggle={onToggle}
+                            onTypeaheadInputChanged={setSearch}
+                            loadingVariant={selectLoadingVariant}
+                            onBlur={() => setSearch('')}
+                            style={{
+                                maxHeight: '275px',
+                                overflowY: 'auto',
+                            }}
```

Comment:
I've noticed us having to do things like this occasionally, to match mocks.

@alanonthegit Let's try to come together with both UX and UI teams for ACS, to pick some standard "customizations" for PatternFly components, and turn them into our own additional classes, so we don't have to be so brittle with style overrides.